app/grants/models.py

```diff
@@ -474,22 +469,24 @@ def get_hash_from_web3(self):
             args['nonce'],
             ).call()
 
-
-    def successful_contribution(self, kwargs):
+    def successful_contribution(self, tx_id):
         """Create a contribution object."""
         from marketing.mails import successful_contribution
         self.last_contribution_date = timezone.now()
-        self.next_contribution_date = timezone.now() + timezone.timedelta(seconds=self.real_period_seconds)
+        self.next_contribution_date = timezone.now() + timezone.timedelta(seconds=int(self.real_period_seconds))
         self.save()
+        # TODO: add gasPrice and nonce
```

Comment:
whats wrong with how this was before?  how do you want to resolve this TODO thats different than how this was beofre?

Reply:
> I do wonder why we save these though. Where do we intend to use them?

 +1 for removing them... i can't recall why we'd need them either